src/app.ts

```diff
@@ -38,6 +41,15 @@ class App {
   private mountRoutes(): void {
     this.app.use(router)
   }
+
+  private async connectDB() {
```

Comment:
why this?

Reply:
directly call createDBConnection in config()

Reply:
To use the async await, i want the database connection to be ready before the server starts listening. Is it incorrect ?

Reply:
No it is not correct. Making DB connection will slow down the server uptime and will lead to performance issues in services which do not require mongo connection.

Reply:
But suppose in some services, we do need database connection and if the database has not been connected. Will that not affect the availability of the service ?

Reply:
Mongoose has inbuild functionality to wait for the query unless db connection is made.

Reply:
Db connection takes time, but what if the database do not get connected at all ? What will happen then ?

Reply:
Server will stop if db connection is not established.

Reply:
Okay. Let me change it.

Reply:
Done.